src/cpuid/src/amd/mod.rs

```diff
@@ -28,6 +28,24 @@ mod registers;
 #[derive(Debug, Clone, Eq, PartialEq)]
 pub struct AmdCpuid(pub std::collections::BTreeMap<CpuidKey, CpuidEntry>);
 
+impl AmdCpuid {
+    /// Include leaves from `other` that are not present in `self`.
+    #[inline]
+    #[must_use]
+    pub fn include_leaves_from(mut self, other: Self) -> Self {
```

Comment:
Don't we have to care about sub-leaves?

Reply:
It would be better to also be able to pass through subleaves but this would require considerable extra logic to be done safely. For now this is safe and should suffice but it may be worth looking into subleaves moving forward.

Reply:
OK, I see. Sorry, but I don't understand why it requires considerable extra logic. Is this not enough?
```
        let leaves = self
            .0
            .iter()
            .map(|x| x.0)
            .collect::<std::collections::HashSet<_>>();

        self.0
            .extend(other.0.into_iter().filter(|x| !leaves.contains(&x.0)));
```

Reply:
The number of sub-leaves within a leaf are denoted using different mechanism. Some may have the length noted in sub-leaf 0, some require getting each sub-leaf until one has a specific value in a specific bit range in a specific register. It is non-trivial to append leaves and it likely requires modification of existing leaves.

Reply:
It's fair. Thanks for detailed explanation!